src/metabase/db/liquibase.clj

```diff
@@ -181,19 +216,34 @@
   (release-lock-if-needed! liquibase)
   (.clearCheckSums liquibase)
   (when (has-unrun-migrations? liquibase)
-   (doseq [^ChangeSet change (.listUnrunChangeSets liquibase nil (LabelExpression.))]
-     (try
-       (log/info (format "Start executing migration with id %s" (.getId change)))
-       ;; Migrate changes from the current state upto the change with id `(.getId change)`
-       ;; Practically, we're running one change at time
-       ;; each .update will be excuted under a transaction, so if it's fails
-       ;; it'll be automatically rollbacked
-       (.update liquibase (.getId change) nil (LabelExpression.))
-       (log/info (u/format-color 'green "[SUCCESS]"))
-       (catch Throwable e
-         ;; if this migration fails, ignore it for the next run
-         (.setIgnore change true)
-         (log/error (u/format-color 'red "[ERROR] %s" (.getMessage e))))))))
+    (let [change-log     (.getDatabaseChangeLog liquibase)
+          fail-on-errors (mapv (fn [^ChangeSet change-set] [change-set (.getFailOnError change-set)])
+                               (.getChangeSets change-log))
+          exec-listener  (proxy [AbstractChangeExecListener] []
+                           (willRun [^ChangeSet change-set _database-change-log _database _run-status]
+                             (when (instance? ChangeSet change-set)
+                               (log/info (format "Start executing migration with id %s" (.getId change-set)))))
+
+                           (runFailed [^ChangeSet change-set _database-change-log _database ^Exception e]
+                             (log/error (u/format-color 'red "[ERROR] %s" (.getMessage e))))
+
+                           (ran [change-set _database-change-log _database ^ChangeSet$ExecType exec-type]
+                             (when (instance? ChangeSet change-set)
+                               (condp = exec-type
+                                 ChangeSet$ExecType/EXECUTED
+                                 (log/info (u/format-color 'green "[SUCCESS]"))
+
+                                 ChangeSet$ExecType/FAILED
+                                 (log/error (u/format-color 'red "[ERROR]"))
+
+                                 (log/info (format "[%s]" (.name exec-type)))))))]
+      (try
+        (doseq [^ChangeSet change-set (.getChangeSets change-log)]
+          (.setFailOnError change-set false))
+        (update-with-change-log liquibase {:exec-listener exec-listener})
```

Comment:
It seems this runs all changesets in a single transaction now with `.run`? Could the comment at the top be updated?
`2. This will attempt to run each migrations one a time`